test/aws/request_test.exs

```diff
@@ -4,43 +4,54 @@ defmodule AWS.RequestTest do
   alias AWS.Client
   alias AWS.Request
 
-  describe "request_rest/9" do
-    defmodule TestClient do
-      @behaviour AWS.HTTPClient
+  @metadata %{
+    api_version: "2014-06-05",
+    content_type: "application/x-amz-json-1.1",
+    endpoint_prefix: "mobileanalytics",
+    global?: false,
+    protocol: "rest-json",
+    signature_version: "v4",
+    signing_name: "mobileanalytics"
+  }
+
+  defmodule TestClient do
+    @behaviour AWS.HTTPClient
+
+    @impl true
+    def request(method, url, body, headers, options) do
+      send(self(), {:request, method, url, body, headers, options})
+
+      {status, _opts} = Keyword.pop(options, :return_status_code, 200)
+
+      {:ok, %{status_code: status, headers: [], body: "{\"Response\":\"foo\"}"}}
+    end
+  end
 
-      @impl true
-      def request(method, url, body, headers, options) do
-        send(self(), {:request, method, url, body, headers, options})
+  describe "request_post/5" do
+    test "do not sign without access_key_id and secret_access_key" do
+      client = %Client{http_client: {TestClient, []}}
 
-        {status, _opts} = Keyword.pop(options, :return_status_code, 200)
+      assert {:ok, _response, _http_response} =
+               Request.request_post(client, @metadata, "Action", %{}, [])
 
-        {:ok, %{status_code: status, headers: [], body: "{\"Response\":\"foo\"}"}}
-      end
+      assert_receive {:request, :post, _url, _body, _headers, _options}
```

Comment:
I think you could match that the `headers` do not have the signature stuff. WDYT?

Reply:
Thanks, I added an assert for the signature not being in headers.